common/src/path.rs

```diff
@@ -1,38 +1,83 @@
 use std::borrow::Cow;
 use std::path::{Path, PathBuf};
 
-fn is_absolute_windows_path(s: &str) -> bool {
-    // UNC
-    if s.len() > 2 && (&s[..2] == "\\\\" || &s[..2] == "//") {
-        return true;
+trait ToChar {
+    fn to_char(self) -> char;
+}
+
+impl ToChar for char {
+    fn to_char(self) -> char {
+        self
     }
+}
 
-    // other paths
-    let mut char_iter = s.chars();
-    let (fc, sc, tc) = (char_iter.next(), char_iter.next(), char_iter.next());
+impl ToChar for u8 {
+    fn to_char(self) -> char {
+        char::from(self)
+    }
+}
 
-    match fc.unwrap_or_default() {
-        'A'..='Z' | 'a'..='z' => {
-            if sc == Some(':') && tc.map_or(true, |tc| tc == '\\' || tc == '/') {
-                return true;
-            }
+impl<T: ToChar + Copy> ToChar for &'_ T {
+    fn to_char(self) -> char {
+        (*self).to_char()
+    }
+}
+
+/// Returns `true` if the given character is any valid directory separator.
+#[inline]
```

Comment:
out of curiosity, do you have any reference on when this is useful?

Reply:
Since it's an internal function, it is most likely inlined anyway during release optimizations, but I'm trying to hint a little more strongly that I want these predicates inlined. I don't have any good reference at the moment, but will look for some.